### ISSUE TITLE
API Unit Tests are silently failing: Permission to write Cloud Trace telemetry

### ISSUE THREAD

**Issue by grayside** [GoogleCloudPlatform/emblem#848]
## Problem

API Unit Tests are silently failing, such as:
* PR: https://github.com/GoogleCloudPlatform/emblem/pull/791
* Build Example: https://console.cloud.google.com/cloud-build/builds/74a703e3-8236-4ad5-b6eb-5d214f364931?project=97614911777

It appears that Content API Staging does not have permission to write telemetry to Cloud Trace.

## Solution

* [ ] Give the emblem-ci-ops service account for Content API the missing permission
* [ ] Modify terraform to give this permission to future service accounts
* [ ] Correct the API Unit Test to fail on permission error, which may be related to running with pytest or shell strict error handling

**Comment by muncus** [comment]
To fix the Cloud Trace issue, we should grant the builder service account `role/cloudtrace.agent` permissions.

**Comment by grayside** [comment]
API Unit Tests pass fine:
```
============================= test session starts ==============================
platform linux -- Python 3.11.3, pytest-6.2.4, py-1.11.0, pluggy-0.13.1
rootdir: /workspace/content-api
collected 7 items

main_approver_test.py ...                                                [ 42%]
main_nonapprover_test.py ....                                            [100%]
```

however, after the tests, a process kicks up to publish OpenTelemetry traces to Cloud Trace. I don't really care what that process is, as I noted above we probably don't want telemetry publishing to happen as part of these tests, but I'm using the fact that it's currently trying to make our CI system more robust.

So the important thing is there's an exception, and Cloud Build isn't aware of it:

```
Traceback (most recent call last):
  File "/builder/home/.local/lib/python3.11/site-packages/opentelemetry/exporter/cloud_trace/__init__.py", line 168, in export
    self.client.batch_write_spans(
  File "/builder/home/.local/lib/python3.11/site-packages/google/cloud/trace_v2/services/trace_service/client.py", line 554, in batch_write_spans
    rpc(
  File "/builder/home/.local/lib/python3.11/site-packages/google/api_core/gapic_v1/method.py", line 113, in __call__
    return wrapped_func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/builder/home/.local/lib/python3.11/site-packages/google/api_core/retry.py", line 349, in retry_wrapped_func
    return retry_target(
           ^^^^^^^^^^^^^
  File "/builder/home/.local/lib/python3.11/site-packages/google/api_core/retry.py", line 191, in retry_target
    return target()
           ^^^^^^^^
  File "/builder/home/.local/lib/python3.11/site-packages/google/api_core/timeout.py", line 120, in func_with_timeout
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/builder/home/.local/lib/python3.11/site-packages/google/api_core/grpc_helpers.py", line 74, in error_remapped_callable
    raise exceptions.from_grpc_error(exc) from exc
google.api_core.exceptions.PermissionDenied: 403 The caller does not have permission
```

The tests are run with `python -m pytest`, but when the exception happens something strange is going on: the exception is not being noted by the shell as an error or crash. This is a clue that `python` CLI is hiding the error from the rest of the system. Typically, tools will have some flag to say "treat errors as errors" if there's really some ambiguity on this, but I don't see anything in https://docs.python.org/3/using/cmdline.html that makes sense to me as that other than the fairly complex options around warning handling.

Is there something that should be done to make sure the process shutdown for pytest has exceptions properly raised and given an actionable exit code from python?

**Comment by pattishin** [comment]
Merged in PR #931 - closing out
